genesis-builder/framework/evm/sources/Evm.move

```diff
@@ -117,6 +117,12 @@ module Evm::evm {
         amount
     }
 
+    // A private function used by the system to call the EVM native.
+    // (For some reason we cannot call the native function directly)
+    fun system_evm_call(caller: address, to: address, value: u256, data: vector<u8>): EvmResult {
```

Comment:
Do we need the owner check just for extra security?

Reply:
If it's a private function there is no way for an external caller to invoke it right? I don't think we also need to check the caller.

Reply:
As long as we don't skip visibility in the execution, this should be safe.